src/server.c

```diff
@@ -641,20 +641,19 @@ int htNeedsResize(dict *dict) {
  * In non cluster-enabled setup, it resize main/expires dictionary based on the same condition described above. */
 void tryResizeHashTables(int dbid) {
     redisDb *db = &server.db[dbid];
-    int slot = 0;
     for (dbKeyType subdict = DB_MAIN; subdict <= DB_EXPIRES; subdict++) {
-        dbIterator *dbit = dbIteratorInitFromSlot(db, subdict, db->sub_dict[subdict].resize_cursor);
-        for (int i = 0; i < CRON_DBS_PER_CALL; i++) {
-            dict *d = dbGetDictFromIterator(dbit);
-            slot = dbIteratorGetCurrentSlot(dbit);
-            dbIteratorNextDict(dbit);
-            if (!d) break;
-            if (htNeedsResize(d))
-                dictResize(d);
+        if (dbSize(db, subdict)) {
```

Comment:
maybe use an early exit (`continue`), instead of indentation?
may also reduce the diff a bit.

Reply:
I personally prefer using the "if + do" approach ; )
And since the iterator has changed, using "continue" can not reduce the diff...

Reply:
not a critical comment, but more and more indentation i usually bad thing (makes things harder to read and maintain). and specifically in cases like this, where you have a job to do on many items in a list but not all, it's better to have an early continue, rather than an if.
this concept breaks if you have several conditional tasks to perform on each item, but that's not the case here, and it won't be in the future, right?